uffi-compat/uffi-compat.lisp

```diff
@@ -599,9 +599,14 @@ output to *trace-output*.  Returns the shell's exit code."
   ;; NULL-TERMINATED-P in CFFI:FOREIGN-STRING-TO-LISP.
   (declare (ignore locale null-terminated-p))
   (let ((ret (gensym)))
-    `(let ((,ret (cffi:foreign-string-to-lisp ,obj
-                                              :count ,length
-                                              :encoding ,encoding)))
+    `(let ((,ret (cffi:foreign-string-to-lisp
+                  ,obj
+                  :count ,length
+                  ;; There are code paths e.g. in clsql leading here
+                  ;; with encoding being nil. UFFI replaces nil
+                  ;; encoding with default foreign encoding in those
+                  ;; cases. Copy that behavior here
+                  :encoding (or ,encoding cffi:*default-foreign-encoding*))))
```

Comment:
I'd drop this comment. It's clear from the code what's going on and your commit message is super clear about why we made this change.

Maybe remove the default value for `encoding`, for simplicity's sake and to avoid expanding to a slightly silly `(or cffi:*default-foreign-encoding* cffi:*default-foreign-encoding*)`.

Reply:
Thanks for the comments!

I removed the comment and the default value for `encoding`.

I also noticed the similar macro `convert-to-foreign-string` had the same issue, so I applied the same fix there as well.